### ISSUE TITLE
[WooMinecraft] null

### ISSUE THREAD

**Issue by beeedee** [WooMinecraft/WooMinecraft#116]
My error (/woo check)

>` [02:30:24 WARN]: [WooMinecraft] null`
> [02:30:24 WARN]: com.plugish.apache.http.client.ClientProtocolException
> [02:30:24 WARN]:        at com.plugish.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:186)
> [02:30:24 WARN]:        at com.plugish.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
> [02:30:24 WARN]:        at com.plugish.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:107)
> [02:30:24 WARN]:        at com.plugish.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:55)
> [02:30:24 WARN]:        at com.plugish.woominecraft.Util.RcHttp.request(RcHttp.java:46)
> [02:30:24 WARN]:        at com.plugish.woominecraft.WooMinecraft.check(WooMinecraft.java:128)
> [02:30:24 WARN]:        at com.plugish.woominecraft.Commands.WooCommand.onCommand(WooCommand.java:31)
> [02:30:24 WARN]:        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44)
> [02:30:24 WARN]:        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:181)
> [02:30:24 WARN]:        at org.bukkit.craftbukkit.v1_7_R4.CraftServer.dispatchCommand(CraftServer.java:765)
> [02:30:24 WARN]:        at org.bukkit.craftbukkit.v1_7_R4.CraftServer.dispatchServerCommand(CraftServer.java:751)
> [02:30:24 WARN]:        at net.minecraft.server.v1_7_R4.DedicatedServer.aB(DedicatedServer.java:344)
> [02:30:24 WARN]:        at net.minecraft.server.v1_7_R4.DedicatedServer.v(DedicatedServer.java:308)
> [02:30:24 WARN]:        at net.minecraft.server.v1_7_R4.MinecraftServer.u(MinecraftServer.java:643)
> [02:30:24 WARN]:        at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:549)
> [02:30:24 WARN]:        at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628)
> [02:30:24 WARN]: Caused by: com.plugish.apache.http.ProtocolException: Target host is not specified
> [02:30:24 WARN]:        at com.plugish.apache.http.impl.conn.DefaultRoutePlanner.determineRoute(DefaultRoutePlanner.java:70)
> [02:30:24 WARN]:        at com.plugish.apache.http.impl.client.InternalHttpClient.determineRoute(InternalHttpClient.java:124)
> [02:30:24 WARN]:        at com.plugish.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:183)
> [02:30:24 WARN]:        ... 15 more

http://begame.epizy.com/shop?wmc_key=8d01thisiscensored
` {"success":true,"data":{"BeeeDee":{"3238":["say %s O art\u0131k vip"]}}}`

my config.yml
```
# Set this to the desired language file you wish to load.
#
# If your l10n is not available but one is that you know how to speak,consider
# contributing to this plugin at https://github.com/WooMinecraft/WooMinecraft/
lang: "en"

# This is how often, in seconds, the server will contact your WordPress installation
# to see if there are donations that need made.
update_interval: 1500

# You must set this to your WordPress site URL.  If you installed WordPress in a
# subdirectory, it should point there.
url: "www.begame.epizy.com/shop"

# This must match the WordPress key in your admin panel for WooMinecraft
# For security purposes, you MUST NOT leave this empty.
key: "8d01thisiscensored"

# Set to true in order to toggle debug information
debug: true
```

version is 1.0.9

**Comment by JayWood** [comment]
Your URL must begin with http for one.

**Comment by beeedee** [comment]
new woo check error

> BeeeDee issued server command: /woo check
> [02:39:44 WARN]: [WooMinecraft] Logging an improper server response
> `[02:39:44 ERROR]: [WooMinecraft] /<html><body><script type="text/javascript" src="/aes.js" ></script><script>function toNumbers(d){var e=[];d.replace(/(..)/g,function(d){e.push(parseInt(d,16))});return e}function toHex(){for(var d=[],d=1==arguments.length&&arguments[0].constructor==Array?arguments[0]:arguments,e="",f=0;f<d.length;f++)e+=(16>d[f]?"0":"")+d[f].toString(16);return e.toLowerCase()}var a=toNumbers("f655ba9d09a112d4968c63579db590b4"),b=toNumbers("98344c2eee86c3994890592585b49f80"),c=toNumbers("fff84cd440dee709fbbf9090b5c6dd43");document.cookie="__test="+toHex(slowAES.decrypt(c,2,a,b))+"; expires=Thu, 31-Dec-37 23:55:55 GMT; path=/"; location.href="http://www.begame.epizy.com/shop?wmc_key=8d0167e7e4d74250a7f0afc2c4aa8436&i=1";</script><noscript>This site requires Javascript to work, please enable Javascript in your browser or use a browser with Javascript support</noscript></body></html>/`
> [02:39:44 WARN]: [WooMinecraft] A JSONObject text must begin with '{' at 1 [character 2 line 1]
> [02:39:44 WARN]: org.json.JSONException: A JSONObject text must begin with '{' at 1 [character 2 line 1]
> [02:39:44 WARN]:        at org.json.JSONTokener.syntaxError(JSONTokener.java:433)
> [02:39:44 WARN]:        at org.json.JSONObject.<init>(JSONObject.java:195)
> [02:39:44 WARN]:        at org.json.JSONObject.<init>(JSONObject.java:319)
> [02:39:44 WARN]:        at com.plugish.woominecraft.WooMinecraft.check(WooMinecraft.java:142)
> [02:39:44 WARN]:        at com.plugish.woominecraft.Commands.WooCommand.onCommand(WooCommand.java:31)
> [02:39:44 WARN]:        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44)
> [02:39:44 WARN]:        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:181)
> [02:39:44 WARN]:        at org.bukkit.craftbukkit.v1_7_R4.CraftServer.dispatchCommand(CraftServer.java:765)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.PlayerConnection.handleCommand(PlayerConnection.java:1047)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.PlayerConnection.a(PlayerConnection.java:884)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.PacketPlayInChat.a(PacketPlayInChat.java:28)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.PacketPlayInChat.handle(PacketPlayInChat.java:65)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.NetworkManager.a(NetworkManager.java:189)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.ServerConnection.c(ServerConnection.java:81)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.MinecraftServer.v(MinecraftServer.java:795)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.DedicatedServer.v(DedicatedServer.java:307)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.MinecraftServer.u(MinecraftServer.java:643)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:549)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628)
> new woo check error
> BeeeDee issued server command: /woo check
> [02:39:44 WARN]: [WooMinecraft] Logging an improper server response
> `[02:39:44 ERROR]: [WooMinecraft] /<html><body><script type="text/javascript" src="/aes.js" ></script><script>function toNumbers(d){var e=[];d.replace(/(..)/g,function(d){e.push(parseInt(d,16))});return e}function toHex(){for(var d=[],d=1==arguments.length&&arguments[0].constructor==Array?arguments[0]:arguments,e="",f=0;f<d.length;f++)e+=(16>d[f]?"0":"")+d[f].toString(16);return e.toLowerCase()}var a=toNumbers("f655ba9d09a112d4968c63579db590b4"),b=toNumbers("98344c2eee86c3994890592585b49f80"),c=toNumbers("fff84cd440dee709fbbf9090b5c6dd43");document.cookie="__test="+toHex(slowAES.decrypt(c,2,a,b))+"; expires=Thu, 31-Dec-37 23:55:55 GMT; path=/"; location.href="http://www.begame.epizy.com/shop?wmc_key=8d01thisiscensored&i=1";</script><noscript>This site requires Javascript to work, please enable Javascript in your browser or use a browser with Javascript support</noscript></body></html>/`
> [02:39:44 WARN]: [WooMinecraft] A JSONObject text must begin with '{' at 1 [character 2 line 1]
> [02:39:44 WARN]: org.json.JSONException: A JSONObject text must begin with '{' at 1 [character 2 line 1]
> [02:39:44 WARN]:        at org.json.JSONTokener.syntaxError(JSONTokener.java:433)
> [02:39:44 WARN]:        at org.json.JSONObject.<init>(JSONObject.java:195)
> [02:39:44 WARN]:        at org.json.JSONObject.<init>(JSONObject.java:319)
> [02:39:44 WARN]:        at com.plugish.woominecraft.WooMinecraft.check(WooMinecraft.java:142)
> [02:39:44 WARN]:        at com.plugish.woominecraft.Commands.WooCommand.onCommand(WooCommand.java:31)
> [02:39:44 WARN]:        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44)
> [02:39:44 WARN]:        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:181)
> [02:39:44 WARN]:        at org.bukkit.craftbukkit.v1_7_R4.CraftServer.dispatchCommand(CraftServer.java:765)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.PlayerConnection.handleCommand(PlayerConnection.java:1047)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.PlayerConnection.a(PlayerConnection.java:884)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.PacketPlayInChat.a(PacketPlayInChat.java:28)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.PacketPlayInChat.handle(PacketPlayInChat.java:65)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.NetworkManager.a(NetworkManager.java:189)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.ServerConnection.c(ServerConnection.java:81)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.MinecraftServer.v(MinecraftServer.java:795)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.DedicatedServer.v(DedicatedServer.java:307)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.MinecraftServer.u(MinecraftServer.java:643)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:549)
> [02:39:44 WARN]:        at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628)

**Comment by JayWood** [comment]
Something on your site is saying `This site requires Javascript to work, please enable Javascript in your browser or use a browser with Javascript support` and is causing your server to not respond correctly.

**Comment by spannerman79** [comment]
`<noscript>This site requires Javascript to work, please enable Javascript in your browser or use a browser with Javascript support</noscript>`

I'd say its a template or plugin thats causing a `noscript` aka no JS fallback - something that I know a default install of Wordpress and WooCommerce doesn't trigger.

@JayWood mark this down as end user issue/outside scope of issue tracker.

**Comment by JayWood** [comment]
@spannerman79 I plan to, just waiting on his response on which theme it is.

**Comment by JayWood** [comment]
@beeedee it's bad practice to do such a thing. The plugin is hooked into `template_redirect` action. If you can tell me which theme/plugin is causing it I can do some testing to see if there's a more suitable hook.

**Comment by beeedee** [comment]
I don't understand

**Comment by JayWood** [comment]
@beeedee Your theme or a plugin is causing that error. In order for a website to function it does not need javascript enabled. Considering the error, I cannot fix this. If you load up a website with javascript disabled you should still be able to view the site, abit broken, but viewable in most cases.

Bottom line, the issue is with a plugin, or your theme.

**Comment by beeedee** [comment]
theme is twenty fourteen
plugin list;
WooCommerce
Minecraft WooCommerce 1.0.9

**Comment by JayWood** [comment]
@beeedee good news is it's not an issue with the plugins or theme. Bad news, it's your host. 

So let me lay out the scenario ( see accompanying GIF below for details ). When someone first comes to your site with no cookies, and no knowledge of your site, they're re-directed to this url:

http://begame.epizy.com/?i=1

This can be accomplished by viewing your site in `incognito` mode in Google Chrome, you'll notice the redirect.

So I did a bit of digging and decided to turn off JavaScript. Once I did that I closed my incognito window which cleared the cookies and all saved data.

I then re-visited your site in incognito mode again ( basically as a new user ) and was greeted with the exact same error, complaining that I had no javascript.

You should know that WordPress itself does NOT throw this error, I say this from 4+ years experience. This issue is with your host.

![fma7deujjz](https://cloud.githubusercontent.com/assets/2303074/19100043/7828e302-8a89-11e6-92a6-d3321c7ea79c.gif)

Overall the issue makes sense to me, your host is calling this javascript for every 'first viewer' which doesn't have the `__test` cookie.

```
function toNumbers(d) {
    var e = [];
    d.replace(/(..)/g, function(d) {
        e.push(parseInt(d, 16))
    });
    return e
}

function toHex() {
    for (var d = [], d = 1 == arguments.length && arguments[0].constructor == Array ? arguments[0] : arguments, e = "", f = 0; f < d.length; f++) e += (16 > d[f] ? "0" : "") + d[f].toString(16);
    return e.toLowerCase()
}
var a = toNumbers("f655ba9d09a112d4968c63579db590b4"),
    b = toNumbers("98344c2eee86c3994890592585b49f80"),
    c = toNumbers("4cb05ee5e16b7664894729737b0a7e44");
document.cookie = "__test=" + toHex(slowAES.decrypt(c, 2, a, b)) + "; expires=Thu, 31-Dec-37 23:55:55 GMT; path=/";
location.href = "http://begame.epizy.com/?i=1";
```

At this point you need to contact your host and ask why ( you're not the first with this redirect ). And if this is a plugin in the `wp-content/mu-plugins/` directory, I might not even be able to override it, depending how it's hooked in.

So to sum it up, this is all on your host at this point. WordPress does not do this by default and this JavaScript issue is not in ANY of the plugins/themes you listed.

**Comment by JayWood** [comment]
@beeedee to further the bad news - https://infinityfree.net/forum/topic/601/ggmake-javascript-not-required-for-acessing-pages

From Infinity Hosting Admin @ Sept 25th
> You can't access a website without Javascript because of a bot protection system.

**Comment by JayWood** [comment]
To that point, there are other hosts out there, but this is the first Free host I've seen in awhile which allows WordPress plugin installations, so your options may be limited.

From a server admin standpoint there are other 'bot protection' systems at the server level out there, it's just a matter of the admin wanting to learn how to use them instead of just installing scripts and calling it a day.
